core/sql/sqlcomp/PrivMgrMD.h

```diff
@@ -130,7 +149,70 @@ typedef struct {
     details += to_string((long long int) objectOwner);
   }
 
-} ObjectReference;
+};
+
+class ObjectUsage
+{
+  public:
+
+  ObjectUsage()
+  : objectUID (0),
+    granteeID (NA_UserIdDefault),
+    grantorIsSystem(false),
+    objectType (COM_UNKNOWN_OBJECT),
+    columnReferences(NULL),
+    originalPrivs(),
+    updatedPrivs()
+  {}
+
+  virtual ~ObjectUsage ( void )
+  {
+    if (columnReferences)
+    {
+      while(!columnReferences->empty())
+        delete columnReferences->back(), columnReferences->pop_back();
+      delete columnReferences;
+    }
+  }
+
+  int64_t objectUID;
+  int32_t granteeID;
+  bool grantorIsSystem;
+  std::string objectName;
+  ComObjectType objectType;
+  std::vector<ColumnReference *> *columnReferences;
+  PrivMgrDesc originalPrivs;
+  PrivMgrDesc updatedPrivs;
+
+  ColumnReference * findColumn (int32_t columnOrdinal)
+  {
+    if (columnReferences == NULL)
+      return NULL;
+    for (int i = 0; i < columnReferences->size(); i++)
+    {
+      ColumnReference *pRef = (*columnReferences)[i];
+      if (pRef->columnOrdinal == columnOrdinal)
```

Comment:
Instead of using a <vector>, you could use a <map> and get direct lookup rather than serial search.